### ISSUE TITLE
ETA in big cities is often not accurate

### ISSUE THREAD

**Issue by oxxm** [osmandapp/Osmand#5773]
Using Osmand routing from one side to the other side of a big city, you get a difference between the calculated ETA and the real arrival time. The latter (real time) is shorter than the calculated ETA sometimes by 10 minutes on a 3km route. One can see how the ETA changes the nearer you get to the destination.

This happens only when driving through a big city, not on highways around it.

Osmand 3.0.4

**Comment by sonora** [comment]
Hard to say if this is really a general error in that direction. I have certainly also seen the opposite effect. Chances are much of it depends on how long the red peroods at stop lights realy are, which is often a regional dependency, how much backup you have at stop lights (which is a question of time of day) and similar. If you feel like researching this further, it would be interesting to know which obstacles you think are overestimated in your city,waiting time at stop lights, left turns, etc.

**Comment by oxxm** [comment]
I tried it with a way from one part of a city to the other, with no turns, just straight route with stop lights. The calculated ETA was the double of the real ETA.

**Comment by vshcherb** [comment]
This bug is related that we don't deal with traffic_signals groups properly, so penalty for stopping at the crossroads could be quite high in some cities

**Comment by aceman444** [comment]
Yes, there are some penalties for traffic signs and pedestrian crossings, which in reality may be driven through faster. It also depends on the city whether the pedestrian crossing and the car traffix sign above it are mapped as a single node instead of two (where OsmAnd would get a penalty twice).
